pkgs/development/libraries/tracker/default.nix

```diff
@@ -29,20 +30,27 @@
 
 stdenv.mkDerivation rec {
   pname = "tracker";
-  version = "3.0.3";
+  version = "3.1.1";
 
   outputs = [ "out" "dev" "devdoc" ];
 
   src = fetchurl {
     url = "mirror://gnome/sources/${pname}/${lib.versions.majorMinor version}/${pname}-${version}.tar.xz";
-    sha256 = "sha256-b1yEqzvh7aUgUBsq7XIhYWoM8VKRDFN3V7U4vAXv/KM=";
+    sha256 = "sha256:1rwafk58dm8fpfdb3yk77g0f9gxkk5dy8hm2yx26y1jlhkly4xj3";
   };
 
   patches = [
     (substituteAll {
       src = ./fix-paths.patch;
       inherit asciidoc;
     })
+    ./fix-docs.patch
+    # Fix 32bit datetime issue, use this upstream patch until 3.1.2 lands
+    # https://gitlab.gnome.org/GNOME/tracker/-/issues/292
+    (fetchpatch {
+      url = "https://gitlab.gnome.org/GNOME/tracker/-/merge_requests/401.diff";
```

Comment:
Note that this patch might change if someone pushes to the branch before it is merged.

Reply:
So I assume it is better to  include the patch in nixpkgs directly for now?

Reply:
Or fetch directly by commit hash

Reply:
I added it now.